### ISSUE TITLE
Create metadata for each repo + app

### ISSUE THREAD

**Issue by slarson** [openworm/OpenWorm#220]

**Comment by gidili** [comment]
Trying to flesh out what we discussed here as the end result:

A web page showing al the repositories under the OpenWorm organisation, grouped by naming convention by default or arbitrarily (e.g. all the org.geppetto repos need to show up as one single entry that can be expanded) and expandanble.

For each repository (or group of repos) the following metadata needs to be shown: 

- link to repository and open issues
- goal of the project (short description)
- inputs and outputs (where applicable)
- connections with other repositories (e.g. Sibernetic output can go to Geppetto and Movement validation)
- Go to person for each repo (name / email)
- Stats
   - Languages used
   - Lines of code
   - etc. useful stuff coming out of the Github API

Metadata for the repos needs to be stored in each repository at the root level. 

A few candidate solutions have been identified:

- Python scripts to interrogate the github API, parse metadata for repos and then output an HTML page, that needs to be deployed / refreshed when stuff changes
- A single page standalone HTML/JS app that generates the view every time it's loaded interrogating the github API and parsing metadata as needed (possibly with a very lite backend depending on needs) 

cc: @tarelli @slarson @adrianq have a look and feel free to add stuff

**Comment by gidili** [comment]
Github raw [allows cross domain](https://jsfiddle.net/John_Idol/wd72ku3d/5/) so there's not even need for backend. Also [RawGit](https://rawgit.com/) mirrors github and serves pages reliably and fast (we had problems before using raw files straight from github).

**Comment by slarson** [comment]
Used a set of tools to produce a schema for the metadata for this:

* Composed target example metadata file as JSON and [extracted JSON schema](http://jsonschema.net/#/)
* [Converted both example and schema to YAML](http://jsontoyaml.com/)
* Verified YAML can be validated using the [JSON Schema Lint validator](http://jsonschemalint.com/draft5/#) (works for equivalent YAML)

Candidate example metadata file:
```
--- 
repo: openworm/PyOpenWorm
shortDescription: "Unified, simple data access python library for data & facts about C. elegans anatomy"
coordinator: stephen@openworm.org
inputs: 
  - openworm/sibernetic
  - openworm/org.geppetto
outputs: 
  - openworm/open-worm-analysis-toolbox
```

Candidate metadata YAML schema:
```
--- 
$schema: "http://json-schema.org/draft-04/schema#"
properties: 
  coordinator: 
    type: string
  inputs: 
    items: 
      type: string
    type: array
  outputs: 
    items: 
      type: string
    type: array
  repo: 
    type: string
  shortDescription: 
    type: string
required: 
  - repo
  - shortDescription
  - inputs
  - outputs
  - coordinator
type: object
```

**Comment by souravsingh** [comment]
I  am  interested in working on the issue.How do I start?

**Comment by gidili** [comment]
Hey @souravsingh sorry for the delayed response and thanks a lot for volunteering :)

We have created metadata for most of the repos and we have a tabbed view of said metadata, this is displayed in a page of the OpenWorm website [here](http://www.openworm.org/repositories.html). 

We are now thinking it would be nice to have an alternative way to visualize this data in tabular form (as a grid or something like that) to show all the metadata of the different repos at a glance. We have documented this in [this issue](https://github.com/openworm/OpenWorm/issues/228). 

Have a look at that issue, and if you're interested we can discuss on the issue and if you have any questions we can do a hangout or something like that might be faster than going back and forth on comments.
